client/src/containers/CampaignAdmin/tests/CampaignAdmin-test.js

```diff
@@ -1,105 +1,210 @@
+  // written after the click. This mean that the test reads sequentially.
+  setTimeout(() => fireEvent.click(loader));
+  mockResponse = {
+    errors: [],
+    record: {
+      id: 1
+    }
+  };
+  await screen.findByTestId('test-breadcrumb');
+  expect(navigate).toBeCalledWith('/campaigns/set/1/show');
+});
 
-    it('detects errors in global messages property', () => {
-      const result = admin.hasErrors({
-        state: {
-          messages: [
-            { value: 'error' },
-          ],
-        },
-      });
-      expect(result).toBe(true);
-    });
+test('CampaignAdmin detects errors in errors property', async () => {
+  const navigate = jest.fn();
+  render(<CampaignAdmin {...makeProps({
+    router: {
+      ...makeProps().router,
+      navigate
+    }
+  })}
+  />);
+  const loader = await screen.findByTestId('test-form-builder-loader');
+  setTimeout(() => fireEvent.click(loader));
+  mockResponse = {
+    errors: [
+      { value: 'error' },
+    ],
+    record: {
+      id: 1
+    }
+  };
+  await screen.findByTestId('test-breadcrumb');
+  expect(navigate).not.toBeCalled();
+});
 
-    it('detects no errors in global messages property', () => {
-      const result = admin.hasErrors({
-        state: {
-          messages: [],
-        },
-      });
-      expect(result).toBe(false);
-    });
+test('CampaignAdmin detects errors global messages property', async () => {
+  const navigate = jest.fn();
+  render(<CampaignAdmin {...makeProps({
+    router: {
+      ...makeProps().router,
+      navigate
+    }
+  })}
+  />);
+  const loader = await screen.findByTestId('test-form-builder-loader');
+  setTimeout(() => fireEvent.click(loader));
+  mockResponse = {
+    errors: [],
+    state: {
+      messages: [
+        { value: 'error' },
+      ]
+    },
+    record: {
+      id: 1
+    }
+  };
+  await screen.findByTestId('test-breadcrumb');
+  expect(navigate).not.toBeCalled();
+});
 
-    it('detects errors in field state field messages', () => {
-      const result = admin.hasErrors({
-        state: {
-          fields: [
-            { name: 'ID' },
-            {
-              name: 'Title',
-              message: { value: 'error' },
-            },
-          ],
-        },
-      });
-      expect(result).toBe(true);
-    });
+test('CampaignAdmin detects no errors in global messages property', async () => {
+  const navigate = jest.fn();
+  render(<CampaignAdmin {...makeProps({
+    router: {
+      ...makeProps().router,
+      navigate
+    }
+  })}
+  />);
+  const loader = await screen.findByTestId('test-form-builder-loader');
+  setTimeout(() => fireEvent.click(loader));
+  mockResponse = {
+    errors: [],
+    state: {
+      messages: []
+    },
+    record: {
+      id: 1
+    }
+  };
+  await screen.findByTestId('test-breadcrumb');
+  expect(navigate).toBeCalledWith('/campaigns/set/1/show');
```

Comment:
Could we test that `no errors in global messages property`?

Reply:
messages is part of a mock response, it makes no sense to check it